corehq/apps/es/management/commands/ensure_indices_reindexed.py

```diff
@@ -0,0 +1,165 @@
+        elif missing_primary and not missing_secondary:
+            user_input = input(
+                f"""It looks like you have followed custom process for reindex.
+                We found that that following old indices are missing from your CommCare setup -
+                {missing_primary}
+                Press 'y' or 'Y' to confirm
+            """)
+
+            if user_input.lower() == 'y':
+                return True
+
+        elif missing_secondary:
+            # Highly unlikely event, adding for some additional safety
+            print(
+                f"""
+                Error: Your Commcare Instance does not seem to be in a consistent state.
+                We found some required indices {missing_secondary} missing from your installation.
+                Please reach out to us on https://forum.dimagi.com/"""
+            )
+            return False
+
+        return False
+
+    @classmethod
+    def _return_missing_index_names(cls):
+        """
+        Returns a tuple of missing primary and secondary index names
+        """
+        missing_primary = []
+        missing_secondary = []
+        for cname in CANONICAL_NAME_ADAPTER_MAP.keys():
+            primary_adapter, secondary_adapter = cls.get_both_adapters_for_cname(cname)
+            if not manager.index_exists(primary_adapter.index_name):
+                missing_primary.append(primary_adapter.index_name)
+            if not manager.index_exists(secondary_adapter.index_name):
+                missing_secondary.append(secondary_adapter.index_name)
+        return missing_primary, missing_secondary
+
+    def is_doc_count_difference_reasonable(self, cname):
+        # We will accept delta of 1% in doc count difference
+        high_frequency_indices = {'case_search', 'cases', 'forms'}
+        primary_adapter, secondary_adapter = self.get_both_adapters_for_cname(cname)
+        delta_perc = self.get_doc_count_delta_percent(cname)
+        if delta_perc == 0:
+            return True
+        elif (
+            primary_adapter.canonical_name in high_frequency_indices
+            and delta_perc <= self.REASONABLE_DIFF_PERCENT
+        ):
+            return True
+        return False
+
+    @classmethod
+    def _primary_index_name(cls, cname):
+        return getattr(const, f'HQ_{cname.upper()}_INDEX_NAME')
+
+    @classmethod
+    def _secondary_index_name(cls, cname):
+        return getattr(const, f'HQ_{cname.upper()}_SECONDARY_INDEX_NAME')
+
+    @classmethod
+    def get_both_adapters_for_cname(cls, cname):
+
+        """
+        returns instance of primary and secondary adapter for the instance
+        """
+        adapter = doc_adapter_from_cname(cname)
+        primary_index_name = cls._primary_index_name(cname)
+        secondary_index_name = cls._secondary_index_name(cname)
+
+        primary_adapter = adapter.__class__(primary_index_name, adapter.type)
+        secondary_adapter = adapter.__class__(secondary_index_name, adapter.type)
```

Comment:
nit: could avoid referencing dunder methods with `type()`
```suggestion
        primary_adapter = type(adapter)(primary_index_name, adapter.type)
        secondary_adapter = type(adapter)(secondary_index_name, adapter.type)
```